source/adapters/hip/common.hpp

```diff
@@ -15,24 +15,39 @@
 #include <hip/hip_runtime.h>
 #include <ur/ur.hpp>
 
-// Hipify doesn't support cuArrayGetDescriptor, on AMD the hipArray can just be
-// indexed, but on NVidia it is an opaque type and needs to go through
-// cuArrayGetDescriptor so implement a utility function to get the array
-// properties
-inline void getArrayDesc(hipArray *Array, hipArray_Format &Format,
-                         size_t &Channels) {
+// Before ROCm 6, hipify doesn't support cuArrayGetDescriptor, on AMD the
+// hipArray can just be indexed, but on NVidia it is an opaque type and needs to
+// go through cuArrayGetDescriptor so implement a utility function to get the
+// array properties
+inline static hipError_t getArrayDesc(hipArray *Array, hipArray_Format &Format,
+                                      size_t &Channels) {
+#if HIP_VERSION_MAJOR >= 6
+  HIP_ARRAY_DESCRIPTOR ArrayDesc;
+  hipError_t err = hipArrayGetDescriptor(&ArrayDesc, Array);
```

Comment:
Do we need to use `hipArrayGetDescriptor` for ROCm 6+ or can we still rely on the previous behaviour?

Reply:
In ROCm 6, `HIP_ARRAY_DESCRIPTOR` is a pointer to a forward-declared type, so once cannot access its fields.